### ISSUE TITLE
Option to show Lightning seed phrase

### ISSUE THREAD

**Issue by Jacksper13** [mynodebtc/mynode#530]
There is currently no way to see your LND wallet's seed phrase. This can be very helpful to check if the phrase is right or in case you lose the phrase but still have access to the node. There are a lot of legitimate reasons why you would want to check this seed phrase after the wallet creation (Samourai offers it in their wallet for example), but there is currently no option to see it in mynode.

There obviously is the safety concern of being able to steal funds this way if someone has access to your node, but at the same time there is an option to pair the wallet behind a password check that essentially gives you access to all the funds in the lightning wallet. I don't see how providing the seed phrase behind a password check is any more dangerous on this regard.

**Comment by tehelsper** [comment]
Yeah, that is a good point regarding the pairing. I'll have to think about it. The RPC is slightly more protected because accessing the RPC is limited by various firewalls and filters in case there was an exploit that could expose the cert key + macaroon. An exploit the read seed data would a bit more serious - I'm sure it's stored within lnd somewhere though, but that is more obfuscated.

I guess I also don't want to encourage people to rely on the node for seed backup or rely on it being there in case of horrible corruption or natural disasters.

A bigger "BACK THIS UP" banner could be helpful as well.

**Comment by Jacksper13** [comment]
Hmm, fair. My concern are the people that just create a wallet to test some of the apps that can only be run with an LND wallet, or just start messing with it without really having a mindset of getting into lightning but then they start digging into it and they can't back up the seed properly. As recently as yesterday we had a case in the TG group.

**Comment by tehelsper** [comment]
Yeah, it is requested relatively often. I wonder if any of the other node options keep a backup of seed phrase.

**Comment by AstroMoonkey** [comment]
Am I understanding the request? Does it assumes the seed phrase is stored in the node? Is the key stored in the node? I do not know, but if it is, where is seed phrase information located. ei. directory?

@taylor Helsper


‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by tehelsper** [comment]
They key is stored somewhere internally within the LND files so it can sign transactions, but it is not easily accessible as a seed phrase as far as I know. The only time to see it is at wallet creation time.

**Comment by raymonddurk** [comment]
I second this. I am not confident which seed is which (I think I have it) but since I am not sure I don't use it and only pair wallets with the node.

**Comment by tehelsper** [comment]
Just FYI, pairing wallets with the lightning node uses the node's key. Re-using the seed phrase should not be done unless recovering the wallet and not to use your node's wallet on your phone (that would not work and may cause channel closure / loss of funds).

**Comment by RR82** [comment]
It is safe to import the LND seed to a BTC wallet, such as Bluewallet (connected to your own Electrum server for privacy). This should show you the onchain funds only (and onchain history), which will verify that it is the same seed. 
Channels are unaffected if you only restore as a BTC wallet. Don't restore as a lightning wallet, that could mess things up!

**Comment by Jacksper13** [comment]
> Yeah, it is requested relatively often. I wonder if any of the other node options keep a backup of seed phrase.

I mean, you have to store the private key somewhere somehow in the node in order to sign transactions...right? Isn't it trivial re-calculating the seed phrase from the private key?

**Comment by 12hourdrive** [comment]
I would like to re-discuss this topic as my node just experienced some issues with the hard drive mounting. I find it vital to view and verify that I have the correct seed. **A mock-recovery,** if you will. Maybe an option in the settings behind the admin password to view the seed phrase, hopefully it's not too much to add.

**Comment by tehelsper** [comment]
This may help. You could verify the first set of addresses match.
https://mynodebtc.github.io/advanced/verify-restore-bluewallet.html

**Comment by 12hourdrive** [comment]
This does help, thank you. I was unaware that MyNode operated this way, only allowing the seedphase to be viewed once, ever. I fell suseptable to this when running the now sunset [Zap Wallet](https://jimmymow.medium.com/announcing-zap-desktop-0-6-0-beta-1724d13ae597) their software also showed the seed phrase once, ever; as well.

In reference to the link you sent, would you recommend taking the node offline, performing the check, and going back online, or simply NOT doing any transactions?

**Comment by tehelsper** [comment]
Just don't do any transactions. It may not show all TXs because the addresses looked up in bluewallet is small, but you can check to see if there was ever a balance on any of them or check to see if any match.
